### DISCUSSION TITLE
Cannot really be installed

### DISCUSSION THREAD

**Discussion by potsky** [spatie/laravel-login-link#2] [Q&A]
Hi !

The idea of the package is very good! And the fact that it can only be installed on `local` env too 👏

In the README: *We highly recommend only installing it as a dev dependency, so this code doesn't wind up in your production environment.*

It works nicely in `local` env but when releasing in staging:

```
2022-05-30 19:14:52.770159473 +0000 UTC [scheduler-1] [2022-05-30 21:14:52] staging.ERROR: Unable to locate a class or view for component [login-link]. {"exception":"[object] (InvalidArgumentException(code: 0): Unable to locate a class or view for component [login-link]. at /app/vendor/laravel/framework/src/Illuminate/View/Compilers/ComponentTagCompiler.php:299)
2022-05-30 19:14:52.770197022 +0000 UTC [scheduler-1] [stacktrace]
2022-05-30 19:14:52.770199104 +0000 UTC [scheduler-1] #0 /app/vendor/laravel/framework/src/Illuminate/View/Compilers/ComponentTagCompiler.php(211): Illuminate\\View\\Compilers\\ComponentTagCompiler->componentClass('login-link')
```

**The provided example with `@env`cannot work** because the tag will still be parsed by Laravel... Any idea without requiring the package globally ?

> ps: in the doc you say *Optionally, you can publish the config file with:* but this is not an option. If you do not publish the config file, the package will be unable to auth the user.

**Comment by freekmurze** [comment]
Ah damn, I didn't think of that. I've updated the readme: the default install instructions don't have that `--dev` flag anymore. 

The package luckily also had a built in way to prevent misuse in other environments.

**Reply by lukasinko** [reply]
@freekmurze Is there anything blocking from not requiring it to install it to "production" composer `require` section ?

**Comment by paulhennell** [comment]
ps: in the doc you say Optionally, you can publish the config file with: but this is not an option. If you do not publish the config file, the package will be unable to auth the user.


Came here to say this. Installed package but it wasn't logging any accounts in, published the config and it works fine. So yeah, not sure it's very optional! ;)

(P.S - Thanks for releasing this! Was just about to make a similar thing for a specific site, then saw you'd released a package 😄 )

**Reply by freekmurze** [reply]
Mmmm, I wonder why I wouldn't work without the config file being published. I'll investigate soon. If you already know a fix, feel free to PR it!

**Comment by WEBronckhorst** [comment]
I use Laravel 11 beta, 
en get this error on composer require:
`
Problem 1
    - spatie/laravel-login-link[0.0.1, ..., 0.0.3, 1.0.0, ..., 1.1.1] require illuminate/contracts ^9.0 -> found illuminate/contracts[v9.0.0-beta.1, ..., 9.x-dev] but these were not loaded, likely because it conflicts with another require.
    - spatie/laravel-login-link[dev-dependabot/github_actions/stefanzweifel/git-auto-commit-action-5, dev-main, 1.1.2, ..., 1.1.3] require illuminate/contracts ^9.0|^10.0 -> found illuminate/contracts[v9.0.0-beta.1, ..., 9.x-dev, v10.0.0, ..., 10.x-dev] but these were not loaded, likely because it conflicts with another require.
    - Root composer.json requires spatie/laravel-login-link * -> satisfiable by spatie/laravel-login-link[dev-dependabot/github_actions/stefanzweifel/git-auto-commit-action-5, dev-main, 0.0.1, 0.0.2, 0.0.3, 1.0.0, ..., 1.1.3, 9999999-dev].

You can also try re-running composer require with an explicit version constraint, e.g. "composer require spatie/laravel-login-link:*" to figure out if any version is installable, or "composer require spatie/laravel-login-link:^2.1" if you know which you need.

`
